feat: Add  option to globally hide the secondary sidebar page TOC (#943)
ansys/ansys-sphinx-theme: doc/source/user-guide/options.rst

Reviewer: Documentation is very clear!

Reviewer 2: ```suggestion
To enable the **What's new** section and sidebar in the changelog file, add the following dictionary
```
GUI elements should be in bold.

Author: Thanks for the insight! That’s new to me, and I’ll make the change.

Reviewer 2: ```suggestion
If a format is used in the ``content`` field that does not fall into the categories above, it is not
```

The dictionary contains the following keys:

> - ``whatsnew_file_path``: The path to the YAML file containing what's new content local to the

Reviewer 2: ```suggestion
- ``whatsnew_file_path``: Path to the YAML file containing what's new content local to the
```
Drop the leading article as we do in API reference doc.

  ``doc/source`` directory. If not provided, the what's new section is not generated.

> - ``changelog_file_path``: The path to the changelog.rst file local to the ``doc/source``

Reviewer 2: ```suggestion
- ``changelog_file_path``: Path to the ``changelog.rst`` file local to the ``doc/source``
```

>   directory. If not provided, the what's new section is not generated.

Reviewer 2: ```suggestion
  directory. If not provided, the **What's new** section is not generated.
```
Referring to UI element so bold.

- ``sidebar_pages``: List of names for the pages to include the what's new sidebar on. If not
  provided, the what's new sidebar is not displayed.

> - ``sidebar_no_of_headers``: Number of minor version sections to display in the what's new sidebar.

Reviewer 2: ```suggestion
- ``sidebar_pages``: List of names for the pages to include in the **What's new** sidebar. If not
 provided, the **What's new** sidebar is not shown.
- ``sidebar_no_of_headers``: Number of minor version sections to display in the **What's new** sidebar.
```

.. note::

    If you are using both the "whatsnew" and "cheatsheet" options, the "cheatsheet" option is
    displayed first in the left navigation pane, followed by the "What's new" section to maintain

>     sidebar consistency.

Reviewer 2: ```suggestion
    If you are displaying both the **What's new** and **Cheat sheet**" section, the **Cheat sheet** section is
   shown first in the left navigation pane, followed by the **What's new** section to maintain
   sidebar consistency.
```

.. warning::

    You must declare the complete layout of the dropdown navigation bar in the YAML file. Sphinx does not resolve it automatically.

Secondary sidebar
-----------------

> The right-hand secondary sidebar shows the **"On this page"** table of contents, which lists

Reviewer: ```suggestion
The right-hand secondary sidebar shows the **"On this page"** table of contents (TOC), which lists
```

the section anchors for the current page. It also contains the *Edit this page* and

> *View source* links.

Reviewer 2: ```suggestion
the section anchors for the current page. It also contains the **Edit this page** and
**View source** links.
```
UI elements -- so bold them.

Hide the page TOC globally
~~~~~~~~~~~~~~~~~~~~~~~~~~

Set ``"show_page_toc": False`` to hide the page TOC on every page.

> The *Edit this page* and *View source* links remain visible:

Reviewer 2: ```suggestion
The **Edit this page** and **View source** links remain visible:
```

.. code:: python

    html_theme_options = {
        "show_page_toc": False,
    }

Show the page TOC inside the primary sidebar
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> When the secondary sidebar TOC is disabled with ``"show_page_toc": False``, you

Reviewer 2: ```suggestion
When the secondary sidebar TOC is turned off by with ``"show_page_toc": False``, you
```

can move the TOC into the primary (left-hand) sidebar instead by setting
``"show_page_toc_in_primary_sidebar": True``.

.. code:: python

    html_theme_options = {
        "show_page_toc": False,
        "show_page_toc_in_primary_sidebar": True,
    }

The TOC entries are injected directly under the current page's entry in the
section navigation tree. A small chevron toggle (``›``) appears beside the page
link. Clicking it expands or collapses the entries manually. As the user

> scrolls through the page, the TOC auto-expands and highlights the active

Reviewer 2: ```suggestion
link. Clicking it expands or collapses the entries manually. As you
scroll through the page, the TOC auto-expands and highlights the active
```

section heading automatically.

Hide the "View source" button globally

> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer 2: ```suggestion
Hide the **View source** button globally
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Set ``"show_source_button": False`` to remove the *View source* link from every page.

> The page TOC and *Edit this page* link remain visible:

Reviewer 2: ```suggestion
Set ``"show_source_button": False`` to remove the **View source*** link from every page.
The page TOC and **Edit this page** link remain visible:
```

.. code:: python

    html_theme_options = {
        "show_source_button": False,
    }

> Both options can be combined:

Reviewer: Having those 2 options disabled remove the left side bar, is it correct or the **edit this page** still appears?

Author: `edit this page`, is configured through the `html_theme_options`. If the option is not defined there, it will not appear. Since it is already included, I did not add it again. I will consider adding it as a note. Nice catch!

.. code:: python

    html_theme_options = {
        "show_page_toc": False,
        "show_source_button": False,
    }

Control the secondary sidebar per page
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

For finer control, use the ``secondary_sidebar_items`` option from
`pydata-sphinx-theme <https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/layout.html#secondary-sidebar-right>`_.

> You can pass a list (applied globally) or a dict with glob patterns (applied per page).

Reviewer 2: In general text, usually a code entity like list would be spelled out.

Author: updated this to 
`You can pass a list of item names (applied globally) or a dictionary with glob patterns as keys (applied per page).`

.. code:: python

    html_theme_options = {
        # Remove the page TOC from all pages but keep other items.
        "secondary_sidebar_items": {
            "**": ["edit-this-page", "sourcelink"],
        },
    }

.. code:: python

    html_theme_options = {
        # Show the page TOC on regular pages; hide it on API and index pages.
        "secondary_sidebar_items": {
            "**": ["page-toc", "edit-this-page", "sourcelink"],
            "api/**": ["edit-this-page", "sourcelink"],
            "index": [],
        },
    }

Control the TOC depth
~~~~~~~~~~~~~~~~~~~~~~

Use ``show_toc_level`` to set how many heading levels are expanded by default in

> the page TOC (default is ``1``, showing only top-level headings):

Reviewer 2: ```suggestion
the page TOC. The default is ``1``, which causes only top-level headings to be shown.
This code would show h2 and h3 headings by default:
```
This seems confusing. Wouldn't a setting of 2 show H1 and H2 headings?

Author: I agree this is a bit confusing. It’s slightly more complicated because H1 doesn’t actually show up in the TOC tree, so a setting of 2 only displays from H2 onward.

.. code:: python

    html_theme_options = {

>         "show_toc_level": 2,  # Show h2 and h3 entries expanded by default

Reviewer 2: ```suggestion
        "show_toc_level": 2,  # Expand h2 and h3 entries by default
```

    }

Hide the page TOC on a single page
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

To suppress the entire secondary sidebar for a single RST page without changing

> ``conf.py``, add the following metadata at the very top of the file:

Reviewer 2: ```suggestion
the ``conf.py` file`, add the following metadata at the very top of the file:
```
All code entities should have a noun describing what type they are.

.. code:: rst

    :html_theme.sidebar_secondary.remove:

>     Page Title

Reviewer 2: ```suggestion
    Page title
```
Given that we use sentence case, lower case the "t"?

Author: yes, it should be. Thank you i will add this.
